crates/base_db/src/fixture.rs

```diff
@@ -333,16 +338,36 @@ struct FileMeta {
     introduce_new_source_root: Option<SourceRootKind>,
 }
 
+fn parse_crate(crate_str: String) -> (String, CrateOrigin) {
+    if let Some((a, b)) = crate_str.split_once("@") {
+        (
+            a.to_owned(),
+            match b.split_once(":") {
+                Some(("CratesIo", data)) => match data.split_once(",") {
+                    Some((version, url)) => CrateOrigin::CratesIo {
+                        name: a.to_owned(),
+                        repo: Some(url.to_owned()),
+                        version: version.to_owned(),
+                    },
+                    _ => panic!("Bad crates.io parameter: {}", data),
+                },
+                _ => panic!("Bad string for crate origin: {}", b),
```

Comment:
Let's return `(crate_str, CrateOrigin::Unknown)` here instead of panicking

Reply:
As this is used in tests, isn't panic better?

Reply:
Ah right, this is part of the fixtures stuff ye then panicking is absolutely fine 👍